XenAdminTests/HealthCheckTests/CredentialTests.cs

```diff
@@ -158,15 +156,91 @@ public void CredentialOp()
             pipeClient.Write(Encoding.UTF8.GetBytes(credential), 0, credential.Length);
             pipeClient.Close();
             System.Threading.Thread.Sleep(1000);
             con = ServerListHelper.instance.GetServerList();
             Assert.IsTrue(con.Count == conSize);
         }
+
+        string metadata = "{\"XenCenter\":{\"System\":{\"Version\":\"0.0.0.0\",\"DotNetVersion\":\"4.0.30319.42000\",\"Culture\":\"English (United States)\"," +
+                           "\"OsVersion\":\"Microsoft Windows NT 6.1.7601 Service Pack 1\",\"OsCulture\":\"English (United States)\",\"IpAddress\":\"\"}," +
+                           "\"Settings\":{\"CFU\":{\"AllowXenCenterUpdates\":true,\"AllowPatchesUpdates\":true,\"AllowXenServerUpdates\":true}," +
+                           "\"Proxy\":{\"UseProxy\":true,\"UseIEProxy\":false,\"BypassProxyForServers\":false,\"ProxyAuthentication\":true,\"ProxyAuthenticationMethod\":\"Digest\"}," +
+                           "\"SaveAndRestore\":{\"SaveSessionCredentials\":true,\"RequireMasterPassword\":false},\"HelpLastUsed\":\"2017-06-09T11:57:49.4046357Z\"}," +
+                           "\"Infrastructure\":{\"TotalConnections\":22,\"Connected\":8},\"Plugins\":[],\"SourceOfData\":\"HealthCheck\",\"Created\":\"2017-07-03 14:24:14Z\"," + 
+                           "\"Reported\":\"@HealthCheckReportTime@\"}}";
+
+        [Test]
+        public void TestMetadata()
+        {
+            // Empty server list
+            ServerListHelper.instance.ClearServerList();
+            int conSize = 0;
+            List<ServerInfo> con = ServerListHelper.instance.GetServerList();
+            Assert.IsTrue(con.Count == conSize);
+
+            // Send metadata, then check that it has been received and processed and that the server list remained unchanged
+            NamedPipeClientStream pipeClient = new NamedPipeClientStream(".", HealthCheckSettings.HEALTH_CHECK_PIPE, PipeDirection.Out);
+            pipeClient.Connect();
+            byte[] metadataBytes = Encoding.UTF8.GetBytes(String.Join(ServerListHelper.SEPARATOR.ToString(), HealthCheckSettings.XENCENTER_METADATA, EncryptionUtils.ProtectForLocalMachine(metadata)));
+            pipeClient.Write(metadataBytes, 0, metadataBytes.Length);
+            pipeClient.Close();
+            System.Threading.Thread.Sleep(1000);
+            con = ServerListHelper.instance.GetServerList();
+            Assert.IsTrue(con.Count == conSize); 
+            Assert.IsTrue(ServerListHelper.instance.XenCenterMetadata == metadata, "\"Send metadata\" test failed"); 
+
+            // Send metadata containing the separator
+            pipeClient = new NamedPipeClientStream(".", HealthCheckSettings.HEALTH_CHECK_PIPE, PipeDirection.Out);
+            pipeClient.Connect();
+            int separatorPosition = new Random().Next(0, metadata.Length);

```

Comment:
I'm a bit worried that using Random without an initial seed makes the test not (or not easily) reproducible. I suppose it is there to bring some randomness into the test -- I was going to suggest pseudo-random number, but that wouldn't really randomize this (because it's being used only once). 

So my suggestion is to keep it as is, but include the random number (position) in the error message.